File: tests/fixtures/Sample.java
package com.example.app;

import java.util.List;
import java.util.Map;
import java.io.*;

public class UserService {
    private static final int MAX_USERS = 100;
    private Map<String, String> cache;

    public UserService() {
        this.cache = new java.util.HashMap<>();
    }

    public String getUser(String id) {
        return cache.get(id);
    }

    private void refreshCache() {
        cache.clear();
    }

    public static UserService create() {
        return new UserService();
    }

    protected int userCount() {
        return cache.size();
    }
}

interface Repository<T> {
    T findById(String id);
    List<T> findAll();
    void save(T entity);
}

public enum Status {
    ACTIVE,
    INACTIVE,
    PENDING;

    public String label() {
        return name().toLowerCase();
    }
}

public record Point(int x, int y) {
    public double distance() {
        return Math.sqrt(x * x + y * y);
    }
}

@interface Cacheable {
    int ttl() default 60;
}

class InternalHelper {
    void doWork() {
        System.out.println("working");
    }
}

File: tests/fixtures/Sample.kt
package com.example.app

import kotlin.collections.List
import kotlin.collections.Map

interface Repository<T> {
    fun findById(id: String): T?
    fun findAll(): List<T>
    fun save(entity: T)
}

data class User(val name: String, val age: Int) {
    fun greet(): String {
        return "Hello, $name!"
    }
}

sealed class Result<out T> {
    data class Success<T>(val data: T) : Result<T>()
    data class Error(val message: String) : Result<Nothing>()
}

enum class Status {
    ACTIVE,
    INACTIVE,
    PENDING;

    fun label(): String = name.lowercase()
}

object AppConfig {
    val version = "1.0.0"

    fun init() {
        println("Initializing...")
    }
}

class UserService(private val repo: Repository<User>) {
    val cache: MutableMap<String, User> = mutableMapOf()

    fun getUser(id: String): User? {
        return cache[id]
    }

    private fun refreshCache() {
        cache.clear()
    }

    companion object {
        fun create(): UserService {
            return UserService(object : Repository<User> {
                override fun findById(id: String): User? = null
                override fun findAll(): List<User> = emptyList()
                override fun save(entity: User) {}
            })
        }
    }
}

fun topLevelFunction(x: Int): Int = x * 2

val topLevelProperty: String = "hello"

typealias StringList = List<String>
